new-lamassu-admin/src/pages/Authentication/Input2FAState.js

```diff
@@ -0,0 +1,123 @@
+import { useMutation, useLazyQuery } from '@apollo/react-hooks'
+import { makeStyles } from '@material-ui/core/styles'
+import gql from 'graphql-tag'
+import React, { useContext, useState } from 'react'
+import { useHistory } from 'react-router-dom'
+
+import AppContext from 'src/AppContext'
+import { Button } from 'src/components/buttons'
+import { CodeInput } from 'src/components/inputs/base'
+import { TL1, P } from 'src/components/typography'
+
+import styles from './shared.styles'
+import { STATES } from './states'
+
+const useStyles = makeStyles(styles)
+
+const INPUT_2FA = gql`
+  mutation input2FA(
+    $username: String!
+    $password: String!
+    $code: String!
+    $rememberMe: Boolean!
+  ) {
+    input2FA(
+      username: $username
+      password: $password
+      code: $code
+      rememberMe: $rememberMe
+    )
+  }
+`
+
+const GET_USER_DATA = gql`
+  {
+    userData {
+      id
+      username
+      role
+    }
+  }
+`
+
+const Input2FAState = ({ state, dispatch }) => {
+  const classes = useStyles()
+  const history = useHistory()
+  const { setUserData } = useContext(AppContext)
+
+  const [invalidToken, setInvalidToken] = useState(false)
+
+  const handle2FAChange = value => {
+    dispatch({
+      type: STATES.INPUT_2FA,
+      payload: {
+        twoFAField: value
+      }
+    })
+    setInvalidToken(false)
+  }
+
+  const handleSubmit = () => {
+    if (state.twoFAField.length !== 6) {
+      setInvalidToken(true)
+      return
+    }
+
+    input2FA({
+      variables: {
+        username: state.clientField,
+        password: state.passwordField,
+        code: state.twoFAField,
+        rememberMe: state.rememberMeField
+      }
+    })
+  }
+
+  const [input2FA, { error: mutationError }] = useMutation(INPUT_2FA, {
+    onCompleted: ({ input2FA: success }) => {
+      success ? getUserData() : setInvalidToken(true)
+    }
+  })
+
+  const [getUserData, { error: queryError }] = useLazyQuery(GET_USER_DATA, {
+    onCompleted: ({ userData }) => {
+      setUserData(userData)
+      history.push('/')
+    }
+  })
```

Comment:
Can you move this to the top of the file? Better to have all hooks together. I think it's also better to invert the order here, since the input2fa one uses getUserData.